cache-hazelcast/src/main/java/io/micronaut/cache/hazelcast/HazelcastFactory.java

```diff
@@ -37,8 +41,35 @@
      * @param clientConfig the configuration read it as a bean
      * @return {@link HazelcastInstance}
      */
+    @Requires(beans = ClientConfig.class)
     @Singleton
-    public HazelcastInstance hazelcastClientInstance(ClientConfig clientConfig) {
+    public HazelcastInstance hazelcastInstance(ClientConfig clientConfig) {
         return HazelcastClient.newHazelcastClient(clientConfig);
     }
+
+    /**
+     * Create a singleton {@link HazelcastInstance} member, based on configuration.
+     *
+     * @param config the configuration read it as a bean
+     * @return {@link HazelcastInstance}
+     */
+    @Requires(beans = Config.class)
+    @Singleton
+    public HazelcastInstance hazelcastInstance(Config config) {
+        return Hazelcast.newHazelcastInstance(config);
+    }
+
+    /**
+     * Create a singleton {@link HazelcastInstance} member or client, if a default Hazelcast configuration resource exists.
+     *
+     * @return {@link HazelcastInstance}
+     */
+    @Requires(missingBeans = {Config.class, ClientConfig.class})
+    @Singleton
+    public HazelcastInstance hazelcastInstance() {
+        if (HazelcastConfigResourceCondition.resourceExists("hazelcast-client.xml", "hazelcast-client.yml")){
```

Comment:
There is a native `@Resources` condition for this:

```java
@Requires(resources="classpath:hazelcast-client.xml")
@Requires(resources="classpath:hazelcast-client.yml")
```

Reply:
Yes, I saw those in the docs. But we need two different beans (client or member implementation of `HazelcastInstance`) by using the same method signature with the different conditions (existence of client or member configuration files). Is there a way to do it using annotations in Micronaut?